apps/website/src/server/trpc/router/donations.ts

```diff
@@ -8,27 +10,108 @@ import {
   DonationSchema,
   type Pixel,
   PixelSchema,
+  type Providers,
 } from "@alveusgg/donations-core";
 
+import { env } from "@/env";
+
 import { sendChatMessage } from "@/server/apis/twitch";
 import {
+  type PixelWithDonation,
   createDonations,
   createPixels,
+  getMyPixels,
+  getPayPalPixelsByVerification,
   getPixels,
   getPublicDonations,
+  renamePixel,
 } from "@/server/db/donations";
+import { triggerPlainDiscordChannelWebhook } from "@/server/outgoing-webhooks";
 import {
+  protectedProcedure,
   publicProcedure,
   router,
   sharedKeyProcedure,
 } from "@/server/trpc/trpc";
+import { limit } from "@/server/utils/rate-limit";
 
 import { channels } from "@/data/twitch";
 
 import { getShortBaseUrl } from "@/utils/short-url";
+import type { trpc } from "@/utils/trpc";
+
+import {
+  PIXEL_RENAME_LOCK_DURATION_MS,
+  pixelIdentifierSchema,
+} from "@/hooks/pixels";
+
+import { payPalVerificationSchema } from "@/components/institute/EditPayPalPixels";
+import { coordsToGridRef } from "@/components/institute/Pixels";
+
+export type RenameMyPixelsMutation = typeof trpc.donations.renameMyPixels;
+export type RenamePayPalPixelsMutation =
+  typeof trpc.donations.renamePayPalPixels;
+export type AnyRenamePixelsMutation =
+  | RenameMyPixelsMutation
+  | RenamePayPalPixelsMutation;
```

Comment:
Don't we normally create these types where we need them by importing the trpc router? This feels almost like a circular definition that these are coming from the trpc util import which presumably ultimately comes from this exact file?